entity-registry/src/main/java/com/linkedin/metadata/aspect/plugins/PluginFactory.java

```diff
@@ -93,6 +159,22 @@ public PluginFactory loadPlugins() {
     return this;
   }
 
+  public boolean isEmpty() {
+    return this.pluginConfiguration.isEmpty()
+        && Optional.ofNullable(this.aspectPayloadValidators).map(List::isEmpty).orElse(true)
+        && Optional.ofNullable(this.mutationHooks).map(List::isEmpty).orElse(true)
+        && Optional.ofNullable(this.mclSideEffects).map(List::isEmpty).orElse(true)
+        && Optional.ofNullable(this.mcpSideEffects).map(List::isEmpty).orElse(true);
+  }
+
+  public boolean hasLoadedPlugins() {
```

Comment:
What if the plugin was loaded with 0 plugins to begin with?

Reply:
The plugin lists expected to either be null (not loaded) or loaded once. I probably need to add some logic to enforce that. With the current code, this is true but it might not be enforced explicitly. I will take a look at this, write the enforcement of that policy and some tests to verify the enforcement.